contents/newsletter/choosing-technologies.md

```diff
@@ -0,0 +1,154 @@
+
+- Expanded our usage of Temporal to power warehouse syncs.
+
+- Moved destinations and transformations away from Node virtual machines to our home-built HogVM in our own language Hog.
+
+- Migrated more of our highest volume services from Python to Rust.
+
+- Evaluated Warpstream and Pulsar as alternatives to Kafka.
+
+- Adopted C++ over Python to speed up SQL parsing. 
+
+We've done all this while shipping multiple new products like web analytics, mobile session replay, and surveys as well as adding features to existing ones. Each of these comes with its own set of technologies to maintain and improve.
+
+Technology is a tool to help us move fast and build something people want. This relies on choosing the **right** technologies as it sets the guidelines of what we can and cannot do.
+
+Although we can't tell you what the right technologies are or the perfect way to pick them, we **can** share the lessons we've learned at PostHog and hope it helps.
+
+## Start with a hair-on-fire problem then find an extinguisher
+
+Adopting new technologies is hard and expensive, so "nice to have" is not enough. We look for hair-on-fire problems, which typically means one of three things:
+
+1. **Costs.** For example, our Infrastructure team does a quarterly review of AWS costs, evaluates where spend is going, and identifies areas to improve it. When costs rise above standard benchmarks, it is time to look into solutions. 
+
+2. **Infrastructure.** For example, reaching scaling limits for our current infrastructure. When adding "more boxes" won't fix the issue, it's time to look for another solution. 
+
+3. **Customers.** For example, customers demand a new feature other companies have like reliable and resumable data exports.
+
+The goal of technology is to solve this hair-on-fire problem. We find potential technologies that do this in one of two ways: 
+
+1. **What are other people using to solve the problem?** Technologies are by definition tools or services other people have used to solve problems. For example, part of the reason [we chose ClickHouse](/blog/how-we-turned-clickhouse-into-our-eventmansion) was that Cloudflare used it.
+
+2. **How have YOU solved this problem in the past?** Pitfalls are often avoidable if you already fell in them once. This means we often rely on our teams' experience with a technology to guide our evaluation.
+
+Realistically, the evaluation process happens at the same time as figuring out the options. This is because some options can be written off immediately as they don't fit our requirements. For example, when we were looking for [scalable replacements for Postgres](/blog/how-we-turned-clickhouse-into-our-eventmansion) in our early days, we knew we needed it to be open source and self-hostable. This immediately cut down our list. 
```

Comment:
```suggestion
Realistically, the evaluation process happens at the same time as figuring out the options. This is because some options can be written off immediately as they don't fit our requirements. When we were looking for [scalable replacements for Postgres](/blog/how-we-turned-clickhouse-into-our-eventmansion) in our early days, for example, we knew we needed it to be open source and self-hostable. This immediately cut down our list. 
```